pkg/kubelet/kubelet_pods.go

```diff
@@ -836,7 +836,6 @@ func containerResourceRuntimeValue(fs *v1.ResourceFieldSelector, pod *v1.Pod, co
 }
 
 // One of the following arguments must be non-nil: runningPod, status.
-// TODO: Modify containerRuntime.KillPod() to accept the right arguments.
```

Comment:
determining the right pod data to pass in kill pod is right here, and kill pod takes the right arguments to handle issues like graceperiod already.